Installing v5.0.1-v5.0.3 throws warnings/errors for uninstalled peer dependencies

After upgrading to v5.0.1 I am getting several warning causes `npm ls` to fail. I am on React v16.2.0 and Jest v22.4.2.

After `npm install`:
```
npm WARN enzyme-adapter-react-13@1.0.3 requires a peer of react@^0.13.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-14@1.0.5 requires a peer of react@^0.14.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-14@1.0.5 requires a peer of react-addons-test-utils@^0.14.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-14@1.0.5 requires a peer of react-dom@^0.14.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15@1.0.5 requires a peer of react@^15.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15@1.0.5 requires a peer of react-dom@^15.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15@1.0.5 requires a peer of react-test-renderer@^15.5.0 but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15.4@1.0.5 requires a peer of react@15.0.0-0 - 15.4.x but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15.4@1.0.5 requires a peer of react-dom@15.0.0-0 - 15.4.x but none is installed. You must install peer dependencies yourself.
npm WARN enzyme-adapter-react-15.4@1.0.5 requires a peer of react-addons-test-utils@15.0.0-0 - 15.4.x but none is installed. You must install peer dependencies yourself.
```

After `npm ls`:
```
├─┬ jest-enzyme@5.0.1
│ ├─┬ enzyme-matchers@5.0.1
│ │ ├── circular-json-es6@2.0.2
│ │ └─┬ deep-equal-ident@1.1.1
│ │   └─┬ lodash.isequal@3.0.4
│ │     ├─┬ lodash._baseisequal@3.0.7
│ │     │ ├── lodash.isarray@3.0.4 deduped
│ │     │ ├── lodash.istypedarray@3.0.6
│ │     │ └── lodash.keys@3.1.2 deduped
│ │     └── lodash._bindcallback@3.0.1
│ ├─┬ enzyme-to-json@3.3.1
│ │ └── lodash@4.17.5 deduped
│ └─┬ jest-environment-enzyme@5.0.1
│   ├── enzyme@3.3.0 deduped
│   ├─┬ enzyme-adapter-react-13@1.0.3
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-14@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-15@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-15.4@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├── enzyme-adapter-react-16@1.1.1 deduped
│   ├── jest-environment-jsdom@22.4.1 deduped
│   ├── UNMET PEER DEPENDENCY react@^0.13.0
│   ├── UNMET PEER DEPENDENCY react-addons-test-utils@^0.14.0
│   ├── UNMET PEER DEPENDENCY react-dom@^0.14.0
│   └── UNMET PEER DEPENDENCY react-test-renderer@^15.5.0
...
npm ERR! peer dep missing: react@^0.13.0, required by enzyme-adapter-react-13@1.0.3
npm ERR! peer dep missing: react@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react@^15.5.0, required by enzyme-adapter-react-15@1.0.5
npm ERR! peer dep missing: react@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-dom@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react-dom@^15.5.0, required by enzyme-adapter-react-15@1.0.5
npm ERR! peer dep missing: react-dom@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-test-renderer@^15.5.0, required by enzyme-adapter-react-15@1.0.5
```

I've  updated to v5.0.2 and run into the same issue:

```
├─┬ jest-enzyme@5.0.2
│ ├─┬ enzyme-matchers@5.0.2
│ │ ├── circular-json-es6@2.0.2
│ │ └─┬ deep-equal-ident@1.1.1
│ │   └─┬ lodash.isequal@3.0.4
│ │     ├─┬ lodash._baseisequal@3.0.7
│ │     │ ├── lodash.isarray@3.0.4 deduped
│ │     │ ├── lodash.istypedarray@3.0.6
│ │     │ └── lodash.keys@3.1.2 deduped
│ │     └── lodash._bindcallback@3.0.1
│ ├─┬ enzyme-to-json@3.3.1
│ │ └── lodash@4.17.5 deduped
│ └─┬ jest-environment-enzyme@5.0.2
│   ├── enzyme@3.3.0 deduped
│   ├─┬ enzyme-adapter-react-13@1.0.3
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-14@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-15@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├─┬ enzyme-adapter-react-15.4@1.0.5
│   │ ├── enzyme-adapter-utils@1.3.0 deduped
│   │ ├── lodash@4.17.5 deduped
│   │ ├── object.assign@4.1.0 deduped
│   │ ├── object.values@1.0.4 deduped
│   │ └── prop-types@15.6.1 deduped
│   ├── enzyme-adapter-react-16@1.1.1 deduped
│   ├── jest-environment-jsdom@22.4.1 deduped
│   ├── UNMET PEER DEPENDENCY react@^0.13.0
│   ├── UNMET PEER DEPENDENCY react-addons-test-utils@^0.14.0
│   ├── UNMET PEER DEPENDENCY react-dom@^0.14.0
│   └── UNMET PEER DEPENDENCY react-test-renderer@^15.5.0
...
npm ERR! peer dep missing: react@^0.13.0, required by enzyme-adapter-react-13@1.0.3
npm ERR! peer dep missing: react@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react@^15.5.0, required by enzyme-adapter-react-15@1.0.5
npm ERR! peer dep missing: react@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-dom@^0.14.0, required by enzyme-adapter-react-14@1.0.5
npm ERR! peer dep missing: react-dom@^15.5.0, required by enzyme-adapter-react-15@1.0.5
npm ERR! peer dep missing: react-dom@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-test-renderer@^15.5.0, required by enzyme-adapter-react-15@1.0.5
```

@blainekasten I am also facing the same issue. I use below config:

    "react": "^15.1.0",
    "enzyme": "^3.2.0",
    "react-addons-test-utils": "^15.3.1",
    "react-dom": "^15.1.0",

And I get below errors:

```
! peer dep missing: react@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-addons-test-utils@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
npm ERR! peer dep missing: react-dom@15.0.0-0 - 15.4.x, required by enzyme-adapter-react-15.4@1.0.5
```

I see in the node_modules/react/package.json that the version installed is 15.6.2 wereas I specifically installed 15.1 as prod dependency.

--- a/package.json
+++ b/package.json
@@ -51,6 +51,7 @@
     "babel-preset-env": "^1.5.2",
     "babel-preset-flow": "^6.23.0",
     "babel-preset-react": "^6.24.1",
+    "enzyme-adapter-react-16": "^1.1.1",
     "eslint": "^4.9.0",
     "eslint-config-airbnb": "^16.1.0",
     "eslint-config-prettier": "^2.9.0",

--- a/packages/jest-environment-enzyme/package.json
+++ b/packages/jest-environment-enzyme/package.json
@@ -39,16 +39,11 @@
     "url": "https://github.com/FormidableLabs/enzyme-matchers/issues"
   },
   "peerDependencies": {
+    "enzyme": "3.x",
     "jest": ">=22.0.0",
     "react": ">=16.x"
   },
   "dependencies": {
-    "enzyme": "^3.0.0",
-    "enzyme-adapter-react-13": "^1.0.3",
-    "enzyme-adapter-react-14": "^1.0.5",
-    "enzyme-adapter-react-15": "^1.0.5",
-    "enzyme-adapter-react-15.4": "^1.0.5",
-    "enzyme-adapter-react-16": "^1.1.1",
     "jest-environment-jsdom": "^22.4.1"
   },
   "babel": {

--- a/packages/jest-environment-enzyme/src/setup.js
+++ b/packages/jest-environment-enzyme/src/setup.js
@@ -1,23 +1,43 @@
 /* eslint-disable global-require */
 
 export const exposeGlobals = () => {
-  let Adapter;
-  switch (global.enzymeAdapterDescriptor) {
+  let dep;
+  switch (global.enzymedepDescriptor) {
     case 'react13':
-      Adapter = require('enzyme-adapter-react-13');
+      dep = 'enzyme-adapter-react-13';
       break;
     case 'react14':
-      Adapter = require('enzyme-adapter-react-14');
+      dep = 'enzyme-adapter-react-14';
       break;
     case 'react15':
-      Adapter = require('enzyme-adapter-react-15');
+      dep = 'enzyme-adapter-react-15';
       break;
     case 'react15.4':
-      Adapter = require('enzyme-adapter-react-15.4');
+      dep = 'enzyme-adapter-react-15.4';
       break;
     case 'react16':
     default:
-      Adapter = require('enzyme-adapter-react-16');
+      dep = 'enzyme-adapter-react-16';
+  }
+
+  let Adapter;
+  try {
+    Adapter = require(dep);
+  } catch (e) {
+    console.error(
+      `
+      Requiring the proper enzyme-adapter during jest-enzyme setup failed.
+      This most likely happens when your application does not properly list the
+      adapter in your devDependencies. Run this line to add the correct adapter:
+
+      > yarn add --dev ${dep}
+
+      or with npm
+      
+      > npm i --save-dev ${dep}
+      `
+    );
+    return;
   }
 
   const { configure, mount, render, shallow } = require('enzyme');

